extension/src/popup/views/SendPayment/index.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState, useEffect } from "react";
+import { useSelector } from "react-redux";
+
+import { Switch } from "react-router-dom";
+import { PublicKeyRoute } from "popup/Router";
+import { ROUTES } from "popup/constants/routes";
+
+import {
+  SendAmount,
+  SendTo,
+  SendSettings,
+  SendConfirm,
+} from "popup/components/sendPayment";
+
+import { getAccountBalances } from "@shared/api/internal";
+import { settingsNetworkDetailsSelector } from "popup/ducks/settings";
+import { publicKeySelector } from "popup/ducks/accountServices";
+
+import { defaultAccountBalances } from "popup/views/Account";
+
+export const SendPayment = () => {
+  // keep state separate for now, combine later if needed
+  const [amount, setAmount] = useState("");
+  const [asset, setAsset] = useState("native");
+  const [destination, setDestination] = useState("");
+  const [transactionFee, setTransactionFee] = useState("");
+  const [memo, setMemo] = useState("");
+
+  const publicKey = useSelector(publicKeySelector);
+  const networkDetails = useSelector(settingsNetworkDetailsSelector);
+  const [accountBalances, setAccountBalances] = useState(
+    defaultAccountBalances,
+  );
+
+  useEffect(() => {
+    const fetchAccountBalances = async () => {
+      try {
+        const res = await getAccountBalances({
+          publicKey,
+          networkDetails,
+        });
+        setAccountBalances(res);
+      } catch (e) {
+        console.error(e);
+      }
+    };
+    fetchAccountBalances();
+  }, [publicKey, networkDetails]);
+
+  // TODO - enforce can't move to next route data not given
+  return (
+    <>
+      <Switch>
+        <PublicKeyRoute exact path={ROUTES.sendPayment}>
+          <SendAmount
+            amount={amount}
+            setAmount={setAmount}
+            asset={asset}
+            setAsset={setAsset}
+            accountBalances={accountBalances}
+          />
+        </PublicKeyRoute>
+        <PublicKeyRoute exact path={ROUTES.sendPaymentTo}>
+          <SendTo destination={destination} setDestination={setDestination} />
+        </PublicKeyRoute>
+        <PublicKeyRoute exact path={ROUTES.sendPaymentSettings}>
+          <SendSettings
+            transactionFee={transactionFee}
+            setTransactionFee={setTransactionFee}
+            memo={memo}
+            setMemo={setMemo}
+          />
+        </PublicKeyRoute>
+        <PublicKeyRoute exact path={ROUTES.sendPaymentConfirm}>
```

Comment:
This route actually needs to be wrapped in `PrivateKeyRoute`. My private key had "expired" in background and I tried to dispatch `signTransactionXDR`. In `signTransactionXDR`, it couldn't find my private key, was unable to sign, and returned an empty string as the signedTransaction (instead of throwing the "Session timed out" error)

Not necessary for this particular PR, but in the next version, we should surface this error

Reply:
oh nice I had a mental note to look into that. I changed them all to be PrivateRoute, I think it would be weird to go through the whole flow only at the confirm step need to re-sign in and go through the flow again
https://github.com/stellar/freighter/pull/372/commits/7ae726e432ad5f766c04fe5ab0296f42aa835749

there's also a weird UX where when you re-sign in you go to the `/sendPayment` page and it's missing the `accountBalances` data so you have to go back to `/account` and then `/sendPayment`. I'll fix that on next iteration